extensions/parquet/table/src/main/java/io/deephaven/parquet/table/PartitionFormatter.java

```diff
@@ -0,0 +1,132 @@
+package io.deephaven.parquet.table;
+
+import io.deephaven.time.DateTimeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class takes an object read from a parquet file and formats it to a string used for generate partitioning column
+ * values.
+ */
+enum PartitionFormatter {
```

Comment:
This is *not* inherently Parquet-specific. Since it's in parquet-table, it *could* be in engine-impl to be adjacent to `PartitionParser`.